lib/AST/Type.cpp

```diff
@@ -4126,10 +4125,6 @@ CanType ProtocolCompositionType::getMinimalCanonicalType(
 
     switch (Req.getKind()) {
     case RequirementKind::Superclass:
-      assert((!superclass || superclass->isEqual(Req.getSecondType()))
-             && "Multiple distinct superclass constraints!");
-      superclass = Req.getSecondType();
```

Comment:
This is a nice simplification but I don't understand why it was failing before. We already skip requirements where the subject type is not equal to the last generic parameter. There should only ever be at most one superclass requirement that reaches this point.

Reply:
The simplification is NFC. 

> This is a nice simplification but I don't understand why it was failing before. 

If the generic signature has more than one requirement, and only one relevant requirement on the last generic parameter, the `Reqs.size() == 1` check does not catch the single-member composition and we hit the corresponding assertion in `ProtocolCompositionType::build`:

```swift
protocol P4 {}
class C: P4 {}
struct S<T: P4> {
  func foo(_: C & P4) {} // Minimal type is C
}
```